bin/generate-gutenboarding-design-thumbnails.js

```diff
@@ -58,72 +90,205 @@ async function run() {
+				for (
+					let //remaining = scrollHeight,
+						screenshotIndex = 0,
+						top = 0;
+					top < scrollHeight;
+					top += snapshotHeight, screenshotIndex++
+				) {
+					// Puppeteer's clip regions get squirrelly when scrolling,
+					// so every shot will be full-sized, and if the last shot
+					// is not an exact multiple we'll adjust it's position in
+					// the final composite
+					const remaining = scrollHeight - top;
+					if ( remaining < snapshotHeight ) {
+						top -= snapshotHeight - remaining;
+					}
+					// const height = Math.min( remaining, snapshotHeight );
+
+					// TODO: add some more smarts here to handle parallax
+					// themes. e.g. the second rockfield_rockfield_rockfield
+					// background image is broken at the seam between two
+					// screenshots because the background image moved between
+					// them.
+					// If this doesn't make sense, have a look at the demo and
+					// think about scrolling down and stiching screenshots
+					// together:
+					// https://public-api.wordpress.com/rest/v1/template/demo/rockfield/rockfield?font_base=Fira%20Sans&font_headings=Playfair%20Display&site_title=Rockfield&language=es
+
+					await page.evaluate( ( top ) => window.scrollTo( { top, behaviour: 'auto' } ), top ); // eslint-disable-line no-shadow
+
+					// Save out intermediate screenshots if debug flag is set
+					const fileName = `${ file }_${ String( screenshotIndex ).padStart( 2, 0 ) }.jpg`;
+					debug && console.log( 'creating', fileName );
+					const maybePathOption = debug ? { path: fileName } : {};
+
+					const newScreenshot = await page.screenshot( {
+						width,
+						height,
+						...maybePathOption,
+						...screenshotOptions,
+					} );
+
+					// hide floating header after the first screenshot to
+					// avoid it being duplicated each frame
+					if ( screenshotIndex === 0 ) {
+						await page.addStyleTag( { content: 'header#masthead{display: none}' } );
+					}
+
+					// Check result matches expectations
+					if ( debug ) {
+						const metadata = await sharp( newScreenshot ).metadata();
+						[ 'width', 'height' ].forEach(
+							( property ) =>
+								metadata[ property ] === eval( property ) ||
+								console.log(
+									`Unexpected ${ property } in screenshot ${ screenshotIndex }. Expected ${ eval(
+										property
+									) }, got ${ metadata[ property ] }`
+								)
+						);
+					}
+
+					screenshots.push( { input: newScreenshot, top, left: 0, width, height } );
+				}
 			} catch ( e ) {
 				if (
 					typeof e.message === 'string' &&
 					e.message.includes( 'Run "npm install" or "yarn install" to download a browser binary.' )
 				) {
 					console.error(
-						'\n\nPlease run `node ./bin/install-gutenboarding-design-thumbnails-dependencies.js` to install the dependencies for this script and then try again.'
+						'\n\nPlease run `PUPPETEER_SKIP_DOWNLOAD= yarn install` to install the chromium binaries required for this script and then try again.'
```

Comment:
Had to add this on my machine:

```suggestion
						'\n\nPlease run `PUPPETEER_PRODUCT=chrome PUPPETEER_SKIP_DOWNLOAD= yarn install` to install the chromium binaries required for this script and then try again.'
```